packages/components/src/tooltip/index.js

```diff
@@ -31,40 +32,83 @@ export const TOOLTIP_DELAY = 700;
 
 const eventCatcher = <div className="event-catcher" />;
 
-const getDisabledElement = ( { eventHandlers, child, childrenWithPopover } ) =>
-	cloneElement(
+/**
+ * Convert a position string to a placement string. This logic
+ * is borrowed from the Popover component, with one key difference:
+ *
+ * Here, the 'left' position translates to the '-start' placement and
+ * the 'right' position translates to the '-end' placement. This ensures
+ * that the ToolTip is placed at the expected left/right location.
+ *
+ * @param {string} position A position string such as 'bottom left'
+ * @return {string} A placement string such as 'bottom-start'
+ */
+const positionToPlacement = ( position ) => {
```

Comment:
This transformation is necessary because otherwise we end up with the following issue (tested in Storybook):

| Before | After |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/14988353/169964975-a5fc19d5-cf2f-4075-97b2-aad88ba9370c.png) | ![image](https://user-images.githubusercontent.com/14988353/169964998-33ede74f-4fac-438d-8f50-c3a5dc13cf20.png) |

Reply:
This [confused me](https://github.com/WordPress/gutenberg/pull/40740#discussion_r869488923) as well.

It's a pity that Popover and Tooltip would have "opposite" behaviour in terms of how `left` and `right` work.